https://www.spinics.net/lists/git/msg417169.html
https://bugs.gentoo.org/751907

The test fails when ${D} is exported in the environment. It actually _shouldn't_
be exported in src_test anyway, but Portage isn't yet fixed for this because
it'll break misbehaving ebuilds (see e.g. pkgcheck's VariableScope warnings).

Upstream are currently discussing other patches and work to find other cases
of environment contamination.
--- a/t/t5582-fetch-negative-refspec.sh
+++ b/t/t5582-fetch-negative-refspec.sh
@@ -105,7 +105,6 @@ test_expect_success "fetch with negative pattern refspec does not expand prefix"
 '

 test_expect_success "fetch with negative refspec avoids duplicate conflict" '
-	cd "$D" &&
 	(
 		cd one &&
 		git branch dups/a &&
--
2.30.2

